src/frontend/src/common/utils/timeUtils.tsx

```diff
@@ -0,0 +1,9 @@
+export const formatTZDate = (d: string): string => {
+  if (!d || d.startsWith("-")) return d;
```

Comment:
I'm assuming the `startsWith("-")` is related to the `UNDEFINED_TEXT` elsewhere, right? This feels kinda magic-value to me, not really clear why `-` should short-circuit the function if you don't know the convention of null being turned into `-`. I'd like to either see a comment that says, "the hyphen is because we use that for undefined/null values" or import `UNDEFINED_TEXT` from somewhere and use that var here (I think it would be clear at that point what's going on).